### PR TITLE
Fixed hasOrderByPrimaryKey evaluation

### PR DESCRIPTION
**Pull request by promatik**
The `$hasOrderByPrimaryKey` was wrong, so order by ID was being added when it wasn't needed.

```sql
select * from `icons` order by `icons.id asc, icons.id DESC limit 40
```

With the fix;
```sql
select * from `icons` order by `icons.id` asc limit 40
```

The function also validates for raw queries.

### CODE REVIEW

File: src/app/Http/Controllers/Operations/ListOperation.php [hunk 1]
```diff
@@ -113,18 +113,12 @@ public function search()
...
-            }
-
-            if ($item['column'] == $this->crud->model->getKeyName()) {
-                $hasOrderByPrimaryKey = true;
+        $modelKey = $this->crud->model->getKeyName();
 
-                return false;
-            }
+        $hasOrderByPrimaryKey = collect($orderBy)->some(function ($item) use ($modelKey) {
+            return isset($item['column']) && $item['column'] === $modelKey || isset($item['sql']) && str_contains($item['sql'], ".$modelKey ");
```
Comment (pxpm) [inline src/app/Http/Controllers/Operations/ListOperation.php:119]: This does not account if you do `$this->crud->query->orderByRaw('id','asc')` it expects `table.id`.

Also what if I `orderByRaw('other_joined_table.id', 'desc')` it will assume that is this table order by primary key that is set ?

Comment (promatik) [reply src/app/Http/Controllers/Operations/ListOperation.php]: @pxpm just pushed a commit to fix the joined tables issue.

Tested with;
- [x] `$this->crud->query->orderByRaw('table.id')`
- [x] `$this->crud->query->orderByRaw('joined_table.id')`
- [x] `$this->crud->query->orderBy('id');`

It fails on `$this->crud->query->orderByRaw('id')`, anyway, this function exists to add a simple improvement to the query, I think we shouldn't add more complexity here, to test cases like:
- `orderByRaw('id')` or 
- `orderByRaw('ideology asc')` or 
- `orderByRaw('joined.id asc, table.id desc')` or 
- `orderByRaw('joined.id, table.ideology')`

The possible cases are a lot, and we would need to carefully parse the raw sql. 
Long story short, the current code covers >95% of the use cases, the other 5% are not even affect by this.

Let me know what you think about this ✌

Comment (pxpm) [reply src/app/Http/Controllers/Operations/ListOperation.php]: 100% agree.

**Comment by pxpm** [comment]
@promatik Just tested your solution and works greate.

Pushed a small fix for an issue reported in gitter when using `global scopes`. The `getQuery()` method does not return the orders applied in global scopes, while `toBase()` does. https://laravel.com/api/8.x/Illuminate/Database/Eloquent/Builder.html#method_toBase

Let me know,
Pedro

**Comment by promatik** [comment]
@pxpm tested the `toBase()` and it's working as expected 👌
I'm going to move this to ready to merge.
